server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceShardsAllocator.java

```diff
@@ -67,12 +73,12 @@ public void allocate(RoutingAllocation allocation) {
         desiredBalanceComputation.onNewInput(
             new DesiredBalanceInput(allocation.immutableClone(), new ArrayList<>(allocation.routingNodes().unassigned().ignored()))
         );
+        listener.onResponse(AcknowledgedResponse.TRUE);// TODO listener need to be passed to the above call
```

Comment:
Could be done after https://github.com/elastic/elasticsearch/pull/86217 is merged

Reply:
It looks like it is not enough to complete this listener once `desiredBalanceComputation` is complete as `processInput` might result in reroute that is async itself.